src/components/modals/cloud-tile.js

```diff
@@ -0,0 +1,182 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import React from 'react';
+import styled from 'styled-components';
+import {Logout, Login} from 'components/common/icons';
+import {
+  CenterVerticalFlexbox,
+  Button
+} from 'components/common/styled-components';
+
+const StyledTileWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: flex-start;
+  border-radius: 2px;
+  border: 1px solid
+    ${props =>
+      props.selected
+        ? props.theme.primaryBtnBgd
+        : props.theme.selectBorderColorLT};
+  color: ${props =>
+    props.selected
+      ? props.theme.primaryBtnBgd
+      : props.theme.selectBorderColorLT};
+  cursor: pointer;
+  font-weight: 500;
+  width: 120px;
+  height: 168px;
+  background-color: #ffffff;
+  transition: ${props => props.theme.transition};
+  position: relative;
+  :hover {
+    border: 1px solid ${props => props.theme.primaryBtnBgd};
+    color: ${props => props.theme.primaryBtnBgd};
+  }
+
+  .button {
+    margin-top: 20px;
+  }
+`;
+
+const CheckMark = styled.span.attrs({
+  className: 'checkbox-inner'
+})`
+  background-color: ${props => props.theme.primaryBtnBgd};
+  position: absolute;
+  bottom: 0;
+  right: 0;
+  display: block;
+  width: 16px;
+  height: 16px;
+  border-top-left-radius: 2px;
+
+  :after {
+    position: absolute;
+    display: table;
+    border: 2px solid #fff;
+    border-top: 0;
+    border-left: 0;
+    transform: rotate(45deg) scale(1) translate(-50%, -50%);
+    opacity: 1;
+    content: ' ';
+    top: 50%;
+    left: 25%;
+    width: 5.7px;
+    height: 9.1px;
+  }
+`;
+const StyledBox = styled(CenterVerticalFlexbox)`
+  margin-right: 12px;
+`;
+
+const StyledCloudName = styled.div`
+  font-size: 12px;
+  margin-top: 12px;
+  margin-bottom: 4px;
+`;
+
+const StyledUserName = styled.div`
+  font-size: 11px;
+  margin-top: 8px;
+  text-align: center;
+  color: ${props => props.theme.primaryBtnActBgd};
+  overflow: hidden;
+  width: 100px;
+  text-overflow: ellipsis;
+`;
+
+const LoginButton = ({onClick}) => (
+  <Button link small onClick={onClick}>
+    <Login />
+    Login
+  </Button>
+);
+
+const LogoutButton = ({onClick}) => (
+  <Button link small onClick={onClick}>
+    <Logout />
+    Logout
+  </Button>
+);
+
+// eslint-disable-next-line complexity
+const CloudTile = ({
+  // action when click on the tile
+  onSelect,
+  // default to login
+  onConnect,
+  // default to logout
+  onLogout,
+  // action name
+  actionName,
+  // cloud provider class
+  cloudProvider,
+  // function to take after login or logout
+  onSetCloudProvider,
+  // whether provider is selected as currentProvider
+  isSelected,
+  // whether user has logged in
+  isConnected
+}) => {
+  const userName =
+    typeof cloudProvider.getUserName === 'function'
+      ? cloudProvider.getUserName()
+      : null;
+  const onClickConnect =
```

Comment:
similar thing here:

```
import {useCallback} from 'react';

const onClickConnect = useCallback(() => typeof onConnect === 'function'
      ? onConnect
      : () => cloudProvider.login(() => onSetCloudProvider(cloudProvider.name))
,[onConnect, cloudProvider.login, cloudProvider.name, onSetCloudProvider]);

```

Reply:
why makes it hard to read, while the performance improvement here is so minimum

Reply:
it will avoid StyledTileWrapper to re-render everytime we render cloud tile